pubtools/_pulp/tasks/garbage_collect.py

```diff
@@ -62,6 +69,22 @@ def run(self):
 
         LOG.info("Temporary repo(s) deletion completed")
 
+        # Clear out old all-rpm-content
+        LOG.info("Start old all-rpm-content deletion")
+        arc_threshold = self.args.arc_threshold
+        criteria = Criteria.and_(
+            Criteria.with_unit_type(RpmUnit),
+            Criteria.with_field("cdn_published", Matcher.less_than(datetime.utcnow() - timedelta(days=arc_threshold))),
+        )
+        try:
+            arc_repo = self.pulp_client.get_repository("all-rpm-content").result()
+            deleted_arc = list(
+                arc_repo.remove_content(criteria=criteria)
+            )
+            LOG.info("Old all-rpm-content deletion completed %s", len(deleted_arc[0].units))
+        except PulpException:
+            LOG.info("all-rpm-content repo unavailable")
```

Comment:
This isn't right, because you don't know if that's the reason for raising PulpException.  If it was raised for any other reason, it's not right to throw away the exception details and output a guess/lie in the logs.

How about changing it like this:

```python
clean_repos = list(self.pulp_client.search_repository(Criteria.with_field("id", "all-rpm-content")))
if not clean_repos:
    LOG.info("No repos found for cleaning.")
    return
arc_repo = clean_repos[0]
# ...then the rest
```

You would tolerate absence of all-rpm-content without silencing any other errors.